Adapters/Vungle/ISVungleAdapter/ISVungleInterstitialAdapterRouter.m

```diff
@@ -0,0 +1,116 @@
+//
+//  ISVungleInterstitialAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleInterstitialAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleInterstitialAdapterRouter()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleInterstitialAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                      parentAdapter:(ISVungleAdapter *)parentAdapter
+                           delegate:(id<ISInterstitialAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _parentAdapter = parentAdapter;
+        _delegate = delegate;
+        _bidPayload = nil;
+    }
+
+    return self;
+}
+
+- (void)loadInterstitial {
+    self.interstitialAd = [[VungleInterstitial alloc] initWithPlacementId:self.placementID];
+    self.interstitialAd.delegate = self;
+    
+    if ([self.interstitialAd canPlayAd]) {
+        LogInternal_Internal(@"Interstitial ad: %@ is loaded", self.placementID);
+        [self.delegate adapterInterstitialDidLoad];
+        return;
+    }
+ 
+    [self.interstitialAd load:self.bidPayload];
+}
+
+- (void)playInterstitialAdWithViewController:(UIViewController *)viewController {
+    [self.interstitialAd presentWith:viewController];
+}
+
+- (void)setBidPayload:(NSString * _Nullable)bidPayload {
+    self.bidPayload = bidPayload;
+}
+
+- (void)interstitialInitSuccess {
+    [self.delegate adapterInterstitialInitSuccess];
+}
+
+- (void)interstitialInitFailed:(NSError *)error;{
+    [self.delegate adapterInterstitialInitFailedWithError:error];
+}
+
+#pragma mark - VungleInterstitialDelegate
+
+- (void)interstitialAdDidLoad:(VungleInterstitial * _Nonnull)interstitial {
+    LogAdapterDelegate_Internal(@"placementID = %@", interstitial.placementId);
+    if (![interstitial canPlayAd]) {
+        // When Interstitial is loaded the canPlayAd should also return YES
+        // If for some reason that is not the case we can also catch it on the Show method
+        LogAdapterDelegate_Internal(@"Vungle Ad is loaded but not ready to be shown");
```

Comment:
Do you need to fail immediately here like in the banner case?

Reply:
@stanley-vungle Hi Stanley, this implementation is also copied 6.x IronSource implementation:
```
if (isAdPlayable && ![self isAdCachedForPlacement:placementID]) {
        // When isAdPlayable is YES the isAdCachedForPlacement should also return YES
        // If for some reason that is not the case we can also catch it on the Show method
        LogAdapterDelegate_Internal(@"Vungle Ad is playable but not ready to be shown");
    }
        
    if (interstitialDelegate) {
        if (isAdPlayable) {
            [interstitialDelegate adapterInterstitialDidLoad];
        } else if (error != nil) {
                [interstitialDelegate adapterInterstitialDidFailToLoadWithError:error];
        }
    }
}
```
In their 6.x implementation, they didn't fail immediately.  They will also check the ad is cached on the Show method. Thanks a lot. :)